ibis/impala/ddl.py

```diff
@@ -774,7 +774,7 @@ def __init__(self, database, like=None, aggregate=False):
         self.like = like
         self.aggregate = aggregate
 
-    def compile(self):
+    def compile(self, params=None):
```

Comment:
It seems like these parameters should all be encapsulated in the DDL object, so passing params into `ibis.impala.compiler.build_ast` is all that's needed.